index.d.ts

```diff
@@ -1,146 +1,148 @@
-    stddev: number,
-    percentiles: Map<number, number>,
-    exceeds: number,
-  }
-
-  enum GCFlag { // eslint-disable-line no-unused-vars
-    // perf_hooks.constants.NODE_PERFORMANCE_GC_FLAGS_NO
-    no = 'no', // eslint-disable-line no-unused-vars
-    // perf_hooks.constants.NODE_PERFORMANCE_GC_FLAGS_CONSTRUCT_RETAINED
-    constructRetained = 'constructRetained', // eslint-disable-line no-unused-vars
-    // perf_hooks.constants.NODE_PERFORMANCE_GC_FLAGS_FORCED
-    forced = 'forced', // eslint-disable-line no-unused-vars
-    // perf_hooks.constants.NODE_PERFORMANCE_GC_FLAGS_SYNCHRONOUS_PHANTOM_PROCESSING
-    synchronousPhantomProcessing = 'synchronousPhantomProcessing', // eslint-disable-line no-unused-vars
-    // perf_hooks.constants.NODE_PERFORMANCE_GC_FLAGS_ALL_AVAILABLE_GARBAGE
-    allAvailableGarbage = 'allAvailableGarbage', // eslint-disable-line no-unused-vars
-    // perf_hooks.constants.NODE_PERFORMANCE_GC_FLAGS_ALL_EXTERNAL_MEMORY
-    allExternalMemory = 'allExternalMemory', // eslint-disable-line no-unused-vars
-    // perf_hooks.constants.NODE_PERFORMANCE_GC_FLAGS_SCHEDULE_IDLE
-    scheduleIdle = 'scheduleIdle' // eslint-disable-line no-unused-vars
+    * Enable/disable specific metrics
+    */
+  collect?: {
+    cpu?: boolean,
+    eventLoopDelay?: boolean,
+    memory?: boolean,
+    gc?: boolean,
+    activeHandles?: boolean
   }
+}
 
-  enum GCOp { // eslint-disable-line no-unused-vars
-    // perf_hooks.constants.NODE_PERFORMANCE_GC_MAJOR
-    major = 'major', // eslint-disable-line no-unused-vars
-    // perf_hooks.constants.NODE_PERFORMANCE_GC_MINOR
-    minor = 'minor', // eslint-disable-line no-unused-vars
-    // perf_hooks.constants.NODE_PERFORMANCE_GC_INCREMENTAL
-    incremental = 'incremental', // eslint-disable-line no-unused-vars
-    // perf_hooks.constants.NODE_PERFORMANCE_GC_WEAKCB
-    weakCB = 'weakCB' // eslint-disable-line no-unused-vars
-  }
+declare interface CPUMetric {
+  /**
+    * Usage percentage
+    */
+  usage: number,
+  /**
+    * Raw value returned by `process.cpuUsage()`
+    */
+  raw: number
+}
 
-  interface GCOpStats {
-    count: number,
-    total: number,
-    average: number
-  }
+/**
+ * On Node 12 and above this is a Histogram instance from 'perf_hooks'.
+ */
+declare interface EventLoopDelayHistogram {
+  min: number,
+  max: number,
+  mean: number,
+  stddev: number,
+  percentiles: Map<number, number>,
+  exceeds: number,
+}
+declare interface EventLoopDelayMetric {
+  /**
+    * computed delay in milliseconds
+    */
+  computed: number,
+  raw: number | EventLoopDelayHistogram
+}
 
-  interface GCAggregatedEntry extends GCOpStats {
-    flags?: Map<GCFlag, GCOpStats>
-  }
+declare enum GCFlag {
+  // perf_hooks.constants.NODE_PERFORMANCE_GC_FLAGS_NO
+  no = 'no', // eslint-disable-line no-unused-vars
+  // perf_hooks.constants.NODE_PERFORMANCE_GC_FLAGS_CONSTRUCT_RETAINED
+  constructRetained = 'constructRetained', // eslint-disable-line no-unused-vars
```

Comment:
Also, to clean this up with `eslint` a little, you can disable the rule and then re-enable it, for example:

```js
/* eslint-disable no-unused-vars */
declare enum GCFlag {
  ...
}
/* eslint-enable no-unused-vars */
```

Reply:
Rather than adding one for every single definition. :)